runtime/runtime/src/tests/apply.rs

```diff
@@ -2793,3 +2837,169 @@ fn test_deploy_and_call_local_receipts() {
         ActionErrorKind::FunctionCallError(FunctionCallError::MethodResolveError(_))
     );
 }
+
+#[test]
+fn test_transaction_ordering_with_apply() {
+    let alice_key1 = InMemorySigner::test_signer(&alice_account());
+    let bob_signer = InMemorySigner::test_signer(&bob_account());
+    let alice_key2 = InMemorySigner::from_seed(alice_account(), KeyType::ED25519, "seed");
+
+    let tx1 = SignedTransaction::send_money(
+        1,
+        alice_account(),
+        alice_account(),
+        &alice_key2, // invalid access key, tx should be dropped
+        100,
+        CryptoHash::default(),
+    );
+    let tx2 = SignedTransaction::send_money(
+        1,
+        alice_account(),
+        alice_account(),
+        &alice_key1,
+        200,
+        CryptoHash::default(),
+    );
+    let tx3 = SignedTransaction::send_money(
+        2,
+        alice_account(),
+        bob_account(),
+        &alice_key1,
+        300,
+        CryptoHash::default(),
+    );
+    let tx4 = SignedTransaction::send_money(
+        1,
+        bob_account(),
+        bob_account(),
+        &bob_signer,
+        400,
+        CryptoHash::default(),
+    );
+    let tx5 = SignedTransaction::send_money(
+        2,
+        bob_account(),
+        alice_account(),
+        &bob_signer,
+        500,
+        CryptoHash::default(),
+    );
+    let tx6 = SignedTransaction::send_money(
+        3,
+        bob_account(),
+        bob_account(),
+        &bob_signer,
+        600,
+        CryptoHash::default(),
+    );
+
+    let txs = vec![tx4.clone(), tx1, tx2.clone(), tx5.clone(), tx3.clone(), tx6.clone()];
+
+    let (runtime, tries, root, apply_state, _signers, epoch_info_provider) = setup_runtime(
+        vec![alice_account(), bob_account()],
+        to_yocto(1_000_000),
+        to_yocto(500_000),
+        10u64.pow(15),
+    );
+
+    let flags = vec![true; txs.len()];
```

Comment:
flags is not a useful name here.  maybe `validity_period`?

Reply:
Decided to rename it to `validity_flags` :)